packages/react/src/components/TagButton/index.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { HTMLProps } from 'react';
+import Icon, { IconType } from '../Icon';
+import { ContentNode } from '../../types';
+import Tag, { TagLabel } from '../Tag';
+import classNames from 'classnames';
+import Button from '../Button';
+
+interface TagButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
+  label: ContentNode;
+  value: ContentNode;
+  icon: IconType;
+  action: (e: React.MouseEvent<HTMLButtonElement>) => void;
+  disabled?: boolean;
+}
+
+const TagButton = ({
+  label,
+  value,
+  icon,
+  action,
+  className,
+  disabled,
+  ...rest
+}: TagButtonProps) => {
+  const accessibilityProps: HTMLProps<HTMLElement> = {};
+
+  if (disabled) {
+    accessibilityProps['aria-disabled'] = disabled;
+  }
+
+  return disabled === undefined ? (
+    <Tag
+      {...rest}
+      {...accessibilityProps}
+      className={classNames('TagButton', 'TagButton-disabled', className)}
+    >
+      <TagLabel>{label}</TagLabel>
+      {value}
+      <Icon className="TagButton__icon" type={icon} />
+    </Tag>
+  ) : (
+    <Button
+      {...rest}
+      variant="tag"
+      className={classNames('TagButton', className)}
+      onClick={action}
+      disabled={disabled}
+    >
+      <TagLabel>{label}</TagLabel>
+      {value}
+      <Icon className="TagButton__icon" type={icon} />
+    </Button>
+  );
```

Comment:
I don't think we should be swapping out the component types here as that can cause issues since it should semantically still be a button in both instances. 

It would be better to always keep it as a button (abbreviated example below):

```jsx
return (
  <Button
    variant="tag"
    disabled={disabled}
    {...props}
  >
    ...
  </Button>
)
```

If the content author also sets `aria-disabled`, we would need to prevent the onclick/action callback from occurring.

Reply:
I was thinking, that statement in the task

> The disabled state is not implicitly supported, the author should instead use Tag in absence of the disabled state.

means that "author" is author of the component, so do we need to remove using of Tag component here?

Reply:
Ah yes, sorry I forgot that requirement.

That's to mean the "author" as in content author and not component author. We don't need to support disabled here because an tag should either be a `TagButton` state or it should not be, and we already have a component that  supports the latter. So I think we should nix `disabled` entirely.